ipaserver/install/plugins/adtrust.py

```diff
@@ -340,43 +358,55 @@ def execute(self, **options):
             logger.debug('AD Trusts are not enabled on this server')
             return False, []
 
-        # Read the gidnumber of the fallback group
-        dn = DN(('cn', ADTRUSTInstance.FALLBACK_GROUP_NAME),
-                self.api.env.container_group,
-                self.api.env.basedn)
-
-        try:
-            entry = ldap.get_entry(dn, ['gidnumber'])
-            gidNumber = entry.get('gidnumber')
-        except errors.NotFound:
-            logger.error("%s not found",
-                         ADTRUSTInstance.FALLBACK_GROUP_NAME)
-            return False, ()
-
+        gidNumber = get_gidNumber(ldap, self.api.env)
         if not gidNumber:
             logger.error("%s does not have a gidnumber",
                          ADTRUSTInstance.FALLBACK_GROUP_NAME)
             return False, ()
 
-        # For each trusted domain object, add gidNumber
+        # For each trusted domain object, add posix attributes
+        # to allow use of a trusted domain account by AD DCs
+        # to authenticate against our Samba instance
         try:
             tdos = ldap.get_entries(
                 DN(self.api.env.container_adtrusts, self.api.env.basedn),
                 scope=ldap.SCOPE_ONELEVEL,
-                filter="(objectclass=ipaNTTrustedDomain)",
-                attrs_list=['gidnumber'])
+                filter="(&(objectclass=ipaNTTrustedDomain)"
+                       "(objectclass=ipaIDObject))",
+                attrs_list=['gidnumber', 'uidnumber', 'objectclass',
+                            'ipantsecurityidentifier',
+                            'ipaNTTrustDirection'
+                            'uid', 'cn', 'ipantflatname'])
             for tdo in tdos:
                 # if the trusted domain object does not contain gidnumber,
                 # add the default fallback group gidnumber
                 if not tdo.get('gidnumber'):
-                    try:
-                        tdo['gidnumber'] = gidNumber
-                        ldap.update_entry(tdo)
-                        logger.debug("Added gidnumber %s to %s",
-                                     gidNumber, tdo.dn)
-                    except Exception:
-                        logger.warning(
-                            "Failed to add gidnumber to %s", tdo.dn)
+                    tdo['gidnumber'] = gidNumber
+
+                # Generate uidNumber and ipaNTSecurityIdentifier if
+                # uidNumber is missing. We rely on sidgen plugin here
+                # to generate ipaNTSecurityIdentifier.
+                if not tdo.get('uidnumber'):
+                    tdo['uidnumber'] = ['-1']
+
+                if not ('posixAccount' in tdo.get('objectclass')):
```

Comment:
``if not ('posixAccount' in tdo.get('objectclass'))`` should be ``if 'posixAccount' not in tdo['objectclass']``.

Reply:
Changed, thanks.